test/runnable/foreach5.d

```diff
@@ -1085,6 +1087,45 @@ void test14653()
 }
 
 /***************************************/
+// 15777
+
+template funA15777()
+{
+    import imports.test15777a;
+    alias funA15777 = fun;
+}
+
+template funB15777()
+{
+    import imports.test15777b;
+    alias funB15777 = fun;
+}
+
+template funAB15777()
+{
+    import imports.test15777a;
+    import imports.test15777b;
+    alias funAB15777 = fun;
+}
+
+void foo15777(alias tpl)()
+{
+    alias seq = AliasSeq!(tpl!());
+    // Make alias of 'overload set' in tuple elements
```

Comment:
Sorry, I was only thinking about the `funAB15777` case. Can you explain what happens there? Shouldn't there be an ambiguity error as the signatures of both functions `foo` are identical? If the two `foo`s were different, would the seq.length equal 2 (when `tpl` is `funAB15777`)?